sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4387,6 +4387,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val OPTIMIZE_UNCORRELATED_IN_SUBQUERIES_IN_JOIN_CONDITION =
```

Comment:
Can you add a comment that this interacts with LEGACY_NULL_IN_EMPTY_LIST_BEHAVIOR